### PR TITLE
Added keyboard shortcut for search box

### LINKED ISSUE #1536: [request] search bar keyboard shortcut
In comparison to Rust's docs.rs, Documenter.jl does not seem to have keyboard shortcuts.

One particularly useful one is being able to press `s` and jumping to the search bar. 

Is this possible?

### PR DESCRIPTION
**Pull request by Hetarth02**
Also, added .vscode to .gitignore
Link to keyboard shortcut is [here](https://github.com/JuliaDocs/Documenter.jl/issues/1536).

To focus into search bar press `ctrl` + `/` or `cmd` + `/` keys.
To focus out of search bar press `Esc` key.

_Edit by @mortenpi: close #1536._

### CODE REVIEW

**Comment by Hetarth02** [comment]
How can I get that one job to be successful?

Is this PR accepted and be merged?

**Review by mortenpi — COMMENTED** [review]
Looks great, thanks!

* I was wondering if we could have a small gray text in the search box, indicating the you can use <kbd>Ctrl + /</kbd> for the search? I think it would greatly help with the discoverability of this feature.

  image

  I guess it should be doable with some simple CSS maybe? But  happy to merge as is too.

* Could you add a brief note into `CHANGELOG.md`, with a reference to this this PR and the related issue?

File: .gitignore [hunk 1]
```diff
@@ -17,3 +17,4 @@ docs/dev/
 docs/build/
 docs/build-pdf/
 docs/site/
+.vscode
```
[the remark this answers is not in the document]
Comment (Hetarth02) [reply .gitignore:20]: - For the time being, would a placeholder like this suffice?

image

I was hoping to revamp the search bar altogether. The tailwind like search bar requires changes on multiple aspects and not only on CSS.

**Comment by mortenpi** [comment]
Ah, yeah, I forgot we show that text. That would be perfect. Probably just `Search docs (ctrl + /)` is sufficient.

**Comment by Hetarth02** [comment]
Alright, will do the changes and make the commit.

**Comment by Hetarth02** [comment]
@mortenpi I have made the suggested changes and some improvements to the method too.

- The new code doesn't require JQuery to work, it is pure Javascript.

**Comment by fredrikekre** [comment]
Why was <kbd>Ctrl</kbd>+<kbd>/</kbd> chosen for this?

**Comment by Hetarth02** [comment]
> Why was <kbd>Ctrl</kbd>+<kbd>/</kbd> chosen for this?

I didn't chose any single Alphanumeric character to avoid triggering the script while still being in the input.(For example, letter `s` to focus in and `e` to focus out. Now we wouldn't want the user to focus in and out while typing the term.)

For this reason, I assumed that a combination of keys would be ideal rather than any single character. Hope it clears up.

**Comment by mortenpi** [comment]
> Why was Ctrl+/ chosen for this?

Fair question. Looking through a few sites, it doesn't look like there is massive consensus about this. GitHub uses just `/`, docs.rs uses just `S`. Vue uses `Ctrl+K`, but that actually apparently clashes with a Chrome shortcut. While I didn't immediately see any prior art for `Ctrl+/`, it seems like a reasonable choice.

File: CHANGELOG.md [hunk 2]
```diff
@@ -34,6 +34,7 @@
 * ![Enhancement][badge-enhancement] Search engine and social media link previews are now supported, with Documenter generating the relevant HTML `meta` tags. ([#1321][github-1321], [#1991][github-1991])
 * ![Enhancement][badge-enhancement] `deploydocs` now supports custom tag prefixes; see section "Deploying from a monorepo" in the docs. ([#1291][github-1291], [#1792][github-1792], [#1993][github-1993])
 * ![Enhancement][badge-enhancement] The `target` keyword of `deploydocs` is now required to point to a subdirectory of `root` (usually the directory where `make.jl` is located). ([#2019][github-2019])
+* ![Enhancement][badge-enhancement] Added keyboard shortcuts for search box (To focus into search box use `ctrl + /` or `cmd + /`, `Esc` to focus out of it). ([#2027][github-2027])
```
Comment (mortenpi) [inline CHANGELOG.md:37]:
```suggestion
* ![Enhancement][badge-enhancement] Added keyboard shortcuts for search box (`ctrl + /` or `cmd + /` to focus into the search box, `Esc` to focus out of it). ([#1536][github-1536], [#2027][github-2027])
```

File: src/html/HTMLWriter.jl [hunk 3]
```diff
@@ -1058,7 +1058,7 @@ function render_sidebar(ctx, navnode)
                 "#documenter-search-query.docs-search-query",
                 :name => "q",
                 :type => "text",
-                :placeholder => "Search docs",
+                :placeholder => "Search Docs (Press 'ctrl + /')",
```
Comment (mortenpi) [inline src/html/HTMLWriter.jl:1061]:
```suggestion
                :placeholder => "Search docs (Press 'ctrl + /')",
```

**Comment by fredrikekre** [comment]
On many keyboards `/` requires shift though, so would be a three button combination.

**Comment by mortenpi** [comment]
We _could_ maybe follow GitHub's lead and use just `/`?

File: src/html/HTMLWriter.jl [hunk 4]
```diff
@@ -1058,7 +1058,7 @@ function render_sidebar(ctx, navnode)
                 "#documenter-search-query.docs-search-query",
                 :name => "q",
                 :type => "text",
-                :placeholder => "Search docs",
+                :placeholder => "Search docs (Press 'ctrl + /')",
```
Comment (mortenpi) [inline src/html/HTMLWriter.jl:1061]:
```suggestion
                :placeholder => "Search docs (ctrl + /)",
```

**Comment by Hetarth02** [comment]
Is there a final combination in mind?

P.S.: It can be changed into whichever combination(s) we like.

File: .gitignore [hunk 5]
```diff
@@ -17,4 +19,4 @@ docs/dev/
 docs/build/
 docs/build-pdf/
 docs/site/
-docs/src/release-notes.md
+docs/src/release-notes.md
```
Comment (mortenpi) [inline .gitignore:22]:
```suggestion
docs/src/release-notes.md

```

File: src/html/HTMLWriter.jl [hunk 6]
```diff
@@ -1058,7 +1058,7 @@ function render_sidebar(ctx, navnode)
                 "#documenter-search-query.docs-search-query",
                 :name => "q",
                 :type => "text",
-                :placeholder => "Search docs",
+                :placeholder => "Search docs (ctrl + /)",
```
Comment (fredrikekre) [inline src/html/HTMLWriter.jl:1061]:
```suggestion
                :placeholder => "Search docs (Ctrl + /)",
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -61,6 +61,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 * The `target` keyword of `deploydocs` is now required to point to a subdirectory of `root` (usually the directory where `make.jl` is located). (#2019)
 
+* Added keyboard shortcuts for search box (`ctrl + /` or `cmd + /` to focus into the search box, `Esc` to focus out of it). (#1536), (#2027)
```
Comment (fredrikekre) [inline CHANGELOG.md:64]:
```suggestion
* Added keyboard shortcuts for search box (`Ctrl + /` or `Cmd + /` to focus into the search box, `Esc` to focus out of it). (#1536), (#2027)
```

**Review by fredrikekre — APPROVED** [review]
Lets try with Ctrl + /. If people prefer something else I think it would be perfectly fine to change this without requiring a breaking Documenter version.

File: .gitignore [hunk 8]
```diff
@@ -1,3 +1,5 @@
+.vscode
```
Comment (fredrikekre) [inline .gitignore:1]: I don't mind that this is added, but note that you can use a global gitignore file for things that are personal (i.e. editor/workflow specific things). See e.g. https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files#configuring-ignored-files-for-all-repositories-on-your-computer.

**Comment by fredrikekre** [comment]
After merging this I have noticed that Ctrl+K seems to be very common. Perhaps Documenter should update to that?

**Comment by mortenpi** [comment]
I would be fine with that. I have also seen that used on some sites.

**Comment by mortenpi** [comment]
Just one more data point: apparently MultiDocumenter sites have a shortcut (to the MultiDocumenter global search) and use `/` (https://github.com/JuliaComputing/MultiDocumenter.jl/issues/45).

**Pull request merged**